annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,322 @@
+        "is_validation": is_validation,
+        "user_id": 1,
+        "job_id": 2,
+    }
+
+    validation_type = ValidationSchema.validation_only
+    db_session.query().filter_by().first().return_value = True
+
+    validate_users_info(db_session, task_info, validation_type)
+
+    assert db_session.query.call_count == 2
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+def test_validate_users_info_calls_check_cross_annotating_pages(
+    mock_session: Mock,
+):
+    db_session = mock_session()
+    task_info = {
+        "is_validation": True,
+        "user_id": 1,
+        "job_id": 2,
+    }
+
+    with patch(
+        f"annotation.tasks.services.{'check_cross_annotating_pages'}"
+    ) as mock_func:
+        validate_users_info(db_session, task_info, ValidationSchema.cross)
+        mock_func.assert_called_once_with(db_session, task_info)
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+@pytest.mark.parametrize(
+    ("is_validation", "validator_or_annotator"),
+    [
+        (True, "validator"),
+        (False, "annotator"),
+    ],
+)
+def test_validate_users_info_raises_error_for_invalid_users_info(
+    mock_session: Mock,
+    is_validation: bool,
+    validator_or_annotator: str,
+):
+    db_session = mock_session()
+    task_info = {
+        "is_validation": is_validation,
+        "user_id": 1,
+        "job_id": 2,
+    }
+    db_session.query().filter_by().first.return_value = None
+
+    expected_error_message = (
+        f"user 1 is not assigned as {validator_or_annotator} for job 2"
+    )
+
+    with pytest.raises(FieldConstraintError, match=expected_error_message):
+        validate_users_info(
+            db_session, task_info, ValidationSchema.validation_only
+        )
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+def test_validate_files_info(mock_session: Mock):
+    db_session = mock_session()
+    task_info = {
+        "file_id": 1,
+        "job_id": 2,
+        "pages": [1, 2, 3],
+    }
+
+    mock_file = Mock(spec=File)
+    mock_file.pages_number = 3
+
+    mock_query = db_session.query.return_value
+    mock_query.filter_by.return_value.first.return_value = mock_file
+
+    validate_files_info(db_session, task_info)
+
+    assert db_session.query.call_count == 1
+    db_session.query.assert_has_calls((call(File),))
+    mock_query.filter_by.assert_called_once_with(file_id=1, job_id=2)
+
+
+@patch("sqlalchemy.orm.Session", spec=True)
+@pytest.mark.parametrize(
+    ("task_info", "mock_file", "expected_error_message_regex"),
+    (
+        (
+            {
+                "file_id": 1,
+                "job_id": 2,
+                "pages": [1, 2, 4],
+            },
+            Mock(spec=File, pages_number=3),
+            r"pages \(\{4\}\) do not belong to file",
+        ),
+        (
+            {
+                "file_id": 1,
+                "job_id": 2,
+                "pages": [1, 2, 3],
+            },
+            None,
+            r"file with id 1 is not assigned for job 2",
```

Comment:
Let's split these cases into two tests. It will simplify this a lot

Reply:
Ok;)